app/assets/javascripts/utils/content/editor/rules-engine.js

```diff
@@ -189,6 +179,8 @@ export default class RulesEngine {
     // new parent is not a layout
     else if (previous.depth === item.depth && !previous.isLayout)
       this.#deny("denyNest");
+    // previous is collapsed, could cause unexpected nesting behaviour
+    else if (previous.hasCollapsedDescendants()) this.#deny("denyNest");
```

Comment:
can we put the `this.#deny("denyNest");` on the next line to be consistent with previous line.
makes it a bit more readable